Source/Dafny/Rewriter.cs

```diff
@@ -2007,6 +2080,22 @@ class Induction_Visitor : BottomUpVisitor {
       }
     }
   }
+
+  class PluginRewriter : IRewriter {
```

Comment:
A Dafny rewriter (IRewriter) can support more phases than a plugin rewriter (which only supports PostResolve)
Therefore, we need to encapsulate any plugin-defined Rewriter into the (more internal) Dafny IRewriter.

Reply:
I would have thought a single abstract class with some virtual methods marked as internal and some as public (the ones we want to expose) would have been sufficient. How come it's not?

Reply:
It would have been sufficient. However, it may not be a good practice for a plugin system because all the "internal" methods would still be in the Plugin namespace, exposed to anyone looking at the source code - which I expect from plugins writers since they need to have their plugin in sync with the Dafny source code.
I prefer to really split what we offer between two things, one for the plugin, and one for internal. That way, if we want to change how the internal work, we are not stuck by a public API that we can always re-route to the new constructs.